minix/lib/libacpi/acpi.c

```diff
@@ -0,0 +1,95 @@
+#define _SYSTEM
+
+#include <errno.h>
+#include <string.h>
+
+#include <minix/acpi.h>
+#include <minix/com.h>
+#include <minix/ds.h>
+#include <minix/ipc.h>
+#include <minix/log.h>
+#include <minix/sysutil.h>
+
+#include "acpi.h"
+
+static struct log log =
+{ .name = "libacpi", .log_level = LEVEL_TRACE, .log_func = default_log };
+
+static endpoint_t acpi_ep = NONE;
+
+int
+do_init()
+{
+	int res;
+	res = ds_retrieve_label_endpt("acpi", &acpi_ep);
+	log_info(&log, "resolved acpi to endpoint: %d\n", acpi_ep);
+	return res;
+}
+
+int
+acpi_init()
```

Comment:
Also, functions without parameters should be declared and defined with (void), not ().